site/content/blog/crossplane-ytt-integration.md

```diff
@@ -0,0 +1,147 @@
+---
+title: "Using ytt to create Crossplane Template Function"
+slug: crossplane-ytt-template-function
+date: 2024-02-05
+author: Rohit Aggarwal
+excerpt: "Using ytt to create composition functions to template Crossplane resources"
+image: /img/logo.svg
+tags: ['Rohit Aggarwal', 'ytt', 'crossplane']
+---
+
+## What is Crossplane?
+
+
+[Crossplane](https://www.crossplane.io/) is an open-source Kubernetes extension that empowers organizations to manage cloud infrastructure across any cloud through standard Kubernetes APIs. It allows platform teams to declaratively define and manage the cloud infrastructure, like  databases, storage volumes, virtual machines, etc., through Kubernetes APIs. 
+
+## Crossplane Composite Resource
+
+A [composite resource](https://docs.crossplane.io/v1.14/concepts/composite-resources/) represents a set of managed resources as a single Kubernetes object. It uses Composition template to create multiple managed resources as a single Kubernetes object. 
+
+## Crossplane Compositions
+
+[Compositions](https://docs.crossplane.io/v1.14/concepts/compositions/) are a template for creating multiple managed resources as a single object. A Composition composes individual managed resources together into a larger, reusable solution. An example of Composition may combine a virtual machine, storage resources, and networking policies. A Composition template links all these individual resources together.
+
+Crossplane Compositions have some limitations though:
+
+1. Compositions don't support conditions, meaning that the transformations they provide are applied on an "all or nothing" basis.
+2. They also don't support loops, which means that you cannot apply transformations iteratively.
+3. Finally, advanced operations are not supported either, like checking for statuses in other systems.
+
+To overcome this, crossplane introduced the composition function.
+
+## Crossplane Composition function
+
+[Composition functions](https://docs.crossplane.io/v1.14/concepts/composition-functions/) (or just functions, for short) are custom programs that template Crossplane resources. Crossplane calls composition functions to determine what resources it should create when you create a composite resource (XR).
+
+## Prerequisites
```

Comment:
Do they need to download all of these in order to use Crossplane? Do they just need to know these tools? Add in more information/instructions for what is required here to proceed further. Also add links to these tools.

Reply:
They need to download all of them. I have added links to install these tools and why they are needed.